src/audio/mux/mux.c

```diff
@@ -486,13 +486,21 @@ SOF_MODULE_INIT(demux, sys_comp_module_demux_interface_init);
 		0xE2, 0xA2, 0xF4, 0x2E, 0x30, 0x69
 SOF_LLEXT_MOD_ENTRY(mux, &mux_interface);
 
-#define UUID_DEMUX 0x68, 0x68, 0xB2, 0xC4, 0x30, 0x14, 0x0E, 0x47, 0x89, 0xA0, \
-		0x15, 0xD1, 0xC7, 0x7F, 0x85, 0x1A
-SOF_LLEXT_MOD_ENTRY(demux, &demux_interface);
+/*
+ * The demux entry is removed because mtl.toml doesn't have an entry
+ * for it. Once that is fixed, the manifest line below can be
+ * re-activated:
+ * #define UUID_DEMUX 0x68, 0x68, 0xB2, 0xC4, 0x30, 0x14, 0x0E, 0x47, 0x89, 0xA0, \
+ *		0x15, 0xD1, 0xC7, 0x7F, 0x85, 0x1A
+ * SOF_LLEXT_MOD_ENTRY(demux, &demux_interface);
+ */
```

Comment:
Do we need an entry for demux for legacy reasons ? if so, can we followup with a PR that adds the demux UUID ?

Reply:
@lgirdwood sure we can restore it if needed, but we need a full TOML entry for it for that, we don't have one ATM

Reply:
ok, pls just copy the mux CPC data for demux and add inline comment. The toml CPC data is in need of tuning in general anyway.

Reply:
@lgirdwood my current demux change is a pure *compile-time* one. Demux is currently disabled in both monolithic and modular IPC4 Intel builds. Also, even for compilation, this change only affects *modular* mux/demux builds, which we don't do either in any default configurations anyway. Whereas if I add a demux TOML entry, it will affect the generated binary, it will add a demux driver, where we haven't got one until now. So I would prefer to do that in a separate PR.